SAT/SATSolver.hpp

```diff
@@ -177,6 +177,13 @@ class SATSolver {
   }
 
   SATLiteralStack explicitlyMinimizedFailedAssumptions(unsigned conflictCountLimit, bool randomize);
+
+  /**
+   * Assuming that the last solving call was unsatisfiable,
+   * try to produce a minimal set of premises that it still unsatisfiable
+   * (with respect to assumptions)
+   */
+  virtual SATClauseList *minimizePremises(SATClauseList *premises) = 0;
```

Comment:
Q: It this is about "last solving call", what are `premises` here? Could we drop them? Or is it more convenient to receive them again here? If so, maybe add a comment they must be the ones from the last call?

Style: Do you really prefer `Type<SPACE>*method_name`? I find it strange, as splits the important, the overall type, with a space. (A tiny bit harder to read for me...)

Reply:
Comment, yes, please.

Thank you for the explanation! (Would never have thought to use a comma when declaring functions/methods; but now I recall the same rule applies to variables. Not too strongly. Just a bit undecided about how much I care about consistency with these things. On the other hand, it's going to be easier for me to recognize what you wrote ;))